ir-engine/etherealengine-archive
# IR-2377 spin products

## Summary
- Ignore drag and only use click in Interactables
- Helper functions for TransformComponent (right, up, forward)
- More comprehensive options for default interaction aliases (spatial, nonspatial, combined - clicks/triggers/E key)
- no longer setting drag to false on mouse up, should reflect state similarly to pressed

- camera orbit system now requires dragging to be true before it captures input and begins orbiting
- InteractableComponent now requires dragging to be false to capture (this allows orbiting if you started mouseDown over an interactable)
- EditorControlSystem no longer needs accumulator or checking mouse down vs up entity, just checks for no dragging on mouse up instead
- no longer starting GizmoPlane with visibleComponent, this is managed in the TransformGizmoControlComponent when a control entity is dragged
- simplified TransformGizmoControlComponent logical flow, now properly updates VisibleComponent on gizmoPlane

## References
paving the way for ecommerce ticket:
[https://tsu.atlassian.net/browse/IR-2377](https://tsu.atlassian.net/browse/IR-2377)

fixes for single-click selection (gizmo plane was interfering)
[https://tsu.atlassian.net/browse/IR-1293](https://tsu.atlassian.net/browse/IR-1293)

## QA Steps
Interactables should only react to a click, not a click & drag (there is a threshold for cursor travel before it is considered a drag)
